store/sqlstore/system_store.go

```diff
@@ -147,3 +147,36 @@ func (s SqlSystemStore) InsertIfExists(system *model.System) (*model.System, err
 	}
 	return system, nil
 }
+
+// UpsertWithCond selects for update a system record with the provided condArgs and updates it's value
+// from the given system arg.
+func (s SqlSystemStore) UpsertWithCond(system *model.System, condArgs map[string]string) (*model.System, error) {
+	tx, err := s.GetMaster().BeginTx(context.Background(), &sql.TxOptions{
+		Isolation: sql.LevelReadCommitted,
```

Comment:
Is this just to capture the additional gap lock? Since Id will anyways be unique, wondering if we need this?

Reply:
I think we need this, for Mysql versions that have auto commit enabled.

Reply:
I'm rusty on this: what's the fundamental difference between an explicit transaction and a `FOR UPDATE` vs. just doing an `UPDATE ... WHERE Value = :Value` and measuring the number of rows changed?